backend/fms_core/viewsets/sample_next_step_by_study.py

```diff
@@ -36,4 +43,25 @@ def destroy(self, request, pk=None):
         if removed:
             return Response(data={"details": removed}, status=status.HTTP_200_OK)
         else:
-            return HttpResponseBadRequest(errors or f"Missing sample-next-step-by-study ID to delete.")
\ No newline at end of file
+            return HttpResponseBadRequest(errors or f"Missing sample-next-step-by-study ID to delete.")
+
+    """
+    Returns the number of samples at each step of a study workflow. A dictionary is returned where
+    the key is a step order ID and the value is the count of the number of samples queued at that step.
+
+    Args:
+    study__id__in: The study ID
+
+    Returns:
+    Dictionary of step order ID / sample count pairs, one for each step in the study workflow.
+    """
+    @action(detail=False, methods=["get"])    
+    def summary(self, request):
```

Comment:
I would rename these endpoints  summary_by_study since they only perform a summary by study. It would be better to have an endpoint that does a full summary (by study) but can optionally take a study_id to filter it to return a single study.

Reply:
Renamed